src/scirpy/ir_dist/metrics.py

```diff
@@ -414,7 +363,362 @@ def _make_numba_matrix(distance_matrix: dict, alphabet: str = "ARNDCQEGHILKMFPST
+        """
+        if len(seqs) == 0 or len(seqs2) == 0:
+            return csr_matrix((len(seqs), len(seqs2)))
+
+        data_rows, indices_rows, row_element_counts = self._metric_mat(
+            seqs=seqs,
+            seqs2=seqs2,
+            is_symmetric=is_symmetric,
+            start_column=start_column,
+        )
+
+        indptr = np.zeros(row_element_counts.shape[0] + 1)
+        indptr[1:] = np.cumsum(row_element_counts)
+        data, indices = np.concatenate(data_rows), np.concatenate(indices_rows)
+        sparse_distance_matrix = csr_matrix((data, indices, indptr), shape=(len(seqs), len(seqs2)))
+        return sparse_distance_matrix
+
+    def calc_dist_mat(self, seqs: Sequence[str], seqs2: Optional[Sequence[str]] = None) -> csr_matrix:
+        """Calculates the pairwise distances between two vectors of gene sequences based on the distance metric
+        of the derived class and returns a CSR distance matrix
+        """
+        if seqs2 is None:
+            seqs2 = seqs
+
+        seqs = np.array(seqs)
+        seqs2 = np.array(seqs2)
+        is_symmetric = np.array_equal(seqs, seqs2)
+
+        if self.n_blocks > 1:
+            split_seqs = np.array_split(seqs, self.n_blocks)
+            start_columns = np.cumsum([0] + [len(seq) for seq in split_seqs[:-1]])
+            arguments = [(split_seqs[x], seqs2, is_symmetric, start_columns[x]) for x in range(self.n_blocks)]
+
+            delayed_jobs = [joblib.delayed(self._calc_dist_mat_block)(*args) for args in arguments]
+            results = joblib.Parallel(return_as="list")(delayed_jobs)
+            distance_matrix_csr = scipy.sparse.vstack(results)
+        else:
+            distance_matrix_csr = self._calc_dist_mat_block(seqs, seqs2, is_symmetric)
+
+        if is_symmetric:
+            upper_triangular_distance_matrix = distance_matrix_csr
+            full_distance_matrix = upper_triangular_distance_matrix.maximum(upper_triangular_distance_matrix.T)
+        else:
+            full_distance_matrix = distance_matrix_csr
+
+        return full_distance_matrix
+
+
+class HammingDistanceCalculator(MetricDistanceCalculator):
+    """Computes pairwise distances between gene sequences based on the "hamming" distance metric.
+
+    The code of this class is based on `pwseqdist <https://github.com/agartland/pwseqdist/blob/master/pwseqdist>`_.
+    Reused under MIT license, Copyright (c) 2020 Andrew Fiore-Gartland.
+
+    Parameters
+    ----------
+    cutoff:
+        Will eleminate distances > cutoff to make efficient
+        use of sparse matrices.
+    n_jobs:
+        Number of numba parallel threads to use for the pairwise distance calculation
+    n_blocks:
+        Number of joblib delayed objects (blocks to compute) given to joblib.Parallel
+    """
+
+    def __init__(
+        self,
+        n_jobs: int = 1,
+        n_blocks: int = 1,
+        cutoff: int = 2,
```

Comment:
I suggest making `normalize` and `histogram` keyword-only. Makes it easier to rearrange the function signature later without making it a breaking API change. 

```suggestion
        cutoff: int = 2,
        *,
```